# ****Midterm of Tomáš Šamaj (36902772)****

In [4]:
%matplotlib inline
import pandas as pd
import seaborn as sns
import math
import matplotlib.pyplot as plt
import scipy.stats as stats
import numpy as np
import seaborn as sns
import datetime
import zipfile

access the data

In [25]:
ADP = pd.read_csv(".\ADP.csv")
EIX = pd.read_csv(".\EIX.csv")
FDS = pd.read_csv(".\FDS.csv")
IPG = pd.read_csv(".\IPG.csv")
ticker_list = [ADP,EIX,FDS,IPG]

#set index to datetime
for ticker in ticker_list:
 ticker.set_index(["Date"],inplace = True)

ADP.head()

High         Low        Open       Close     Volume  \
Date                                                                    
2017-11-13  112.059998  110.550003  110.980003  111.529999  3716200.0   
2017-11-14  112.230003  110.940002  111.309998  112.080002  1992000.0   
2017-11-15  112.230003  110.379997  112.199997  110.660004  1555200.0   
2017-11-16  111.480003  110.470001  111.000000  111.029999  1870200.0   
2017-11-17  111.220001  110.269997  110.980003  111.059998  1969500.0   

             Adj Close  
Date                    
2017-11-13  100.551643  
2017-11-14  101.047508  
2017-11-15   99.767288  
2017-11-16  100.100868  
2017-11-17  100.127922

1) Is there a company that has no difference between the Close and Adj Close columns?
What does it mean from the financial point of view for the stock (you can get bonus
partial points)?

In [44]:
print(sum(ADP["Close"]==ADP["Adj Close"])==0)
print(sum(EIX["Close"]==EIX["Adj Close"])==0)
print(sum(FDS["Close"]==FDS["Adj Close"])==0)
print(sum(IPG["Close"]==IPG["Adj Close"])==0)


False
False
False
False


Since it is not true for any ticker, there is a difference between Close and Adj Close columns for all companies.

From the financial point of view this means that during the sample period, each of the 4 companies has at least once paid dividends/split stocks or issued new shares after trading hours. In such cases, the Adj. Close price is different from the Close price of the day.

2) What is the highest and lowest price (Adj Close) each company recorded?

In [54]:
for ticker in ticker_list:
    print(ticker["Adj Close"].max(),ticker["Adj Close"].min())

259.8173217773437 98.1278076171875
72.75 40.61245727539063
487.8295593261719 178.72164916992188
39.13483428955078 11.541755676269531


ADP: max=259.8173217773437, min=98.1278076171875
EIX: max=72.75, min=40.61245727539063
FDS: max=487.8295593261719, min=178.72164916992188
IPG: max=39.13483428955078, min=11.541755676269531